### ISSUE TITLE
[SC2153] impossible possible misspelling when unbound variables stop program 

### ISSUE THREAD

**Issue by bruce-ricard** [koalaman/shellcheck#2796]
#### feature suggestion
- Rule Id SC2153: 

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

set -o nounset

unset=foo

echo "$UNSET $unset"
```

#### Here's what shellcheck currently says:

> echo "$UNSET $unset"
      ^----^ SC2153 (info): Possible misspelling: UNSET may not be assigned. Did you mean unset?

#### Here's what I wanted or expected to see:

When `set -o nounset` is there, IMHO this warning shouldn't be shown, as there already is a mechanism in place to catch it.
The [exceptions](https://github.com/koalaman/shellcheck/wiki/SC2153#exceptions) section of the documentation states that if the undefined variable has a mechanism in place to catch it, then the warning doesn't appear. IMO it is the case when `nounset` is set.
